drivers/timer/arcv2_timer0.c

```diff
@@ -384,11 +394,7 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 		delay -= unannounced;
 		delay = MAX(delay, MIN_DELAY);
 
-		if (delay > MAX_CYCLES) {
-			last_load = MAX_CYCLES;
-		} else {
-			last_load = delay;
-		}
+		last_load = MIN(delay, MAX_CYCLES);
```

Comment:
I'd suggest to move it to a separate change as well as it has nothing to do with the main reason for this patch - yet another subtle yet quite fundamental redesign of Timer IRQ handling for ARC.

Reply:
It's already in different commit